bool full
bool recharge_needed

